lib/archethic/networking/ip_lookup.ex

```diff
@@ -29,6 +29,20 @@ defmodule Archethic.Networking.IPLookup do
     ip
   end
 
+  defp is_public_ip(ip, provider) do
+    case Networking.valid_ip?(ip) do
+      true ->
+        Logger.info("Node IP discovered by #{provider}")
+        ip
+
+      false ->
+        case provider == Strict do
```

Comment:
I would suggest to put it in the `else` of the `with` statement, it will be easier to read.